---
title: "didactic-spork (P26201 - cultures)"
author: "George"
date: '2022-07-12'
output:
  bookdown::html_document2:
    toc: true
    toc_float: true
    toc_depth: 3
    number_sections: true
    theme: spacelab
    highlight: textmate
    df_print: paged
    code_folding: hide
    self_contained: false
    keep_md: false
    encoding: "UTF-8"
---

```{r themes}
source("/Users/geweaa/Box Sync/Protocols/theme.R")
```

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, cache = TRUE, fig.path = "/figures")
library(tidyverse)
```


### Read files and assign colors


```{r files}
seqtab <- read_tsv("data/ASV_table.tsv", col_types = cols(.default = col_character(), 
                                    count = col_integer()
                                    )) 

# Use GTDB release R07-RS207 throughout the analysis (version 207)
# Cite https://doi.org/10.17044/scilifelab.14869077.v4
# P15009 (Sci-life contract id M.Dopson_19_05) was sampled for cultures and feeding experiment
# P21015 (Sci-life contract id M.Dopson_21_01) was sampled for Larwence Berkeley National Laboratory
# P15009 was sampled Oct 2019, P21015 was sampled Feb 2020

tax <- read_tsv("data/ASV_tax.tsv", col_types = cols(.default = col_character())
                ) %>% 
  mutate(phylum = gsub("_[A-Z]", "", phylum)) %>%
  select(-kingdom, -confidence) %>%
  mutate(taxon = coalesce(genus, family, order, class, phylum, domain))
smd <- read_tsv("data/smd.tsv", col_types = cols(.default = col_character())
                ) %>% 
  filter(groundwater %in% c("KR0015","SA1420","SA2600")) 

seqtab %>% 
  inner_join(smd, by = "sample") %>% 
  filter(age == "ctrl" & medium == "ace") %>% 
  pull(seqid) %>% unique() -> ace # Contaminant ASVs in acetate medium
seqtab %>% 
  inner_join(smd, by = "sample") %>% 
  filter(age == "ctrl" & medium == "lys") %>% 
  pull(seqid) %>% unique() -> lys # Contaminant ASVs in lysate medium

seqtab %>%
  inner_join(smd, by = "sample") %>%
  filter(
    age != "ctrl" |
    medium == "env" |
    substr(sample,1,6) == "P21015" & medium == "ace" & !seqid %in% ace |
    substr(sample,1,6) == "P21015" & medium == "lys" & !seqid %in% lys |
    substr(sample,1,6) == "P26201"
    ) %>%
  group_by(sample) %>% mutate(relab = count/sum(count)) %>% ungroup() %>%
  select(sample,seqid,count,relab) -> seqtab
```

This project includes 15 metagenomes from 15 unique cultures

```{r files metag}
coverm <- data.frame() # Soon will contain 35 de-replicated genomes

for (file in list.files("data", pattern = ".gz", full.names = TRUE)) {
  name <- substr(file, 6, nchar(file)) %>% gsub("_L002_1.trim.fastq.gz", "", .)
  i <- read_tsv(file, show_col_types = F) %>% rename(genome = 1, tpm = 2)
  i <- i %>% mutate(sample = name) %>% filter(tpm > 0) %>% select(genome, sample, tpm)
  coverm <- coverm %>% rbind(i)
}

# Sample 35 (l7f) was removed from the binning process as this sample did not produce any bins
bintax <- read_tsv("data/gtdbtk.summary.tsv", col_types = cols(.default = col_character())) %>% 
  filter(genome %in% coverm$genome)
checkm <- read_tsv("data/bin-stats-checkm.tsv", show_col_types = F) %>%
  filter(genome %in% coverm$genome)
```


```{r define colours}
cols.gw <- c(
  "KR0015" = "#D8B70A",
  "SA1420" = "#A2A475",
  "SA2600" = "#1B5331"
  )

cols.fraction <- c(
  "Environmental" = "#D8B70A",
  "0.1 - 0.45 µm" = "#A2A475",
  "> 0.1 µm" = "#1B5331"
  )

cols.mag <- c(
  "Other" = "#899DA4",
  "Nanoarchaeota" = "#046C9A",
  "Patescibacteria" = "#C7B19C",
  "Proteobacteria" = "#972D15",
  "Desulfobacterota" = "#A2A475",
  "Firmicutes" = "#02401B"
  )

cols.phylum <- c( # Sorted on abundance
  "Patescibacteria" = "#C7B19C",
  "Desulfobacterota" = "#A2A475",
  "Caldatribacteriota" = "#972D15",
  "Chloroflexota" = "#D69C4E",
  "Acidobacteriota" = "#1B5331",
  "Omnitrophota" = "#00A08A",
  "Nitrospirota" = "#D8B70A",
  "Campylobacterota" = "#FAEFD1",
  "Other" = "#899DA4",
  "Unidentified" = "#046C9A"
)
```


### Alpha diversity ###

Estimate alpha diversity using the Shannon-Weaver index on species level

```{r alpha diversity}
seqtab %>%
  select(-relab) %>%
  spread(seqid, count, fill = 0) %>% 
  column_to_rownames('sample') %>%
  vegan::diversity() %>% as.data.frame() %>%
  rownames_to_column('sample') %>%
  rename(shannon = 2) %>%
  inner_join(smd, by = "sample") -> adiv
```


```{r plot alpha diversity, fig.width = 10/2.53}
adiv %>%
  mutate(fraction = if_else(fraction == 'filtered',
                            '0.1 - 0.45 µm',
                            '> 0.1 µm')
         ) %>%
  replace_na(list("fraction" = "Environmental")) %>%
  ggplot(aes(x = fct_relevel(groundwater, c("KR0015","SA1420","SA2600")),
             y = shannon)) +
  geom_boxplot(aes(fill = fct_relevel(fraction, names(cols.fraction)), 
                   colour = fraction), 
               outlier.colour = "white", alpha = 0.5
               ) +
  labs(x = "", y = "Alpha diversity (Shannon H index)", fill = "") +
  theme_tidy(legend = c(0.85,0.88)) +
  scale_fill_manual(values = cols.fraction) +
  scale_colour_manual(values = cols.fraction, guide = "none") +
  theme( 
    legend.margin = margin(),
    legend.background = element_rect(fill = "white"),
    panel.grid.major.y = element_line(colour = "grey", linetype = "dotted", linewidth = 0.4))
```


```{r export alphadiv}
ggsave(filename = "figures/adiv.pdf", width = 10, height = 10, units = "cm")
```


### NMDS cultures ###

```{r nmds unfiltered}
set.seed(999)
nmds <- seqtab %>% # Full size fraction
  inner_join(smd, by = "sample") %>%
  filter(fraction == "unfiltered" | is.na(fraction)) %>%
  group_by(seqid) %>% filter(count > 2) %>% ungroup() %>%
  select(seqid, sample, count) %>% spread(seqid, count, fill = 0) %>%
  column_to_rownames("sample") %>% vegan::metaMDS()

vegan::scores(nmds)$sites %>%
  as.data.frame() %>%
  rownames_to_column("sample") %>%
  inner_join(smd, by = "sample") -> nmds.scores
```

Plot the ordination

```{r plot nmds unfiltered, include = T}
p1 <- nmds.scores %>%
  mutate(medium = if_else(medium == "env","Environmental","Culture")) %>%
  # Plot
  ggplot(aes(x = NMDS1, y = NMDS2)) +
  geom_vline(xintercept = 0, linetype = 'dotted') +
  geom_hline(yintercept = 0, linetype = 'dotted') +
  # Points for samples, coloured by nature
  geom_point(aes(colour = groundwater, shape = medium), size = 3, stroke = 1) +
  theme_tidy() +
  scale_shape_manual(values = c("Culture" = 1, "Environmental" = 19), name = "") +
  scale_color_manual(values = cols.gw, name = "Groundwater") +
  annotate('text', x = -Inf, y = -Inf, size = 7/.pt, 
           label = paste('Stress = ', round(nmds$stress, digits = 2)),
           hjust = -0.1, vjust = -1,
           ) +
  ggtitle("> 0.1 µm fraction") +
  theme(
    plot.title = element_text(hjust = .5, size = 7)
    )
```


```{r extract nmds coordinates filtered}
nmds <- seqtab %>% # Small size fraction
  inner_join(smd, by = "sample") %>%
  filter(fraction == "filtered" | is.na(fraction)) %>%
  group_by(seqid) %>% filter(count > 2) %>% ungroup() %>%
  select(seqid, sample, count) %>% spread(seqid, count, fill = 0) %>%
  column_to_rownames("sample") %>% vegan::metaMDS()

vegan::scores(nmds)$sites %>%
  as.data.frame() %>%
  rownames_to_column("sample") %>%
  inner_join(smd, by = "sample") -> nmds.scores
```


```{r plot nmds filtered}
p2 <- nmds.scores %>%
  mutate(medium = if_else(medium == "env","Environmental","Culture")) %>%
  # Plot
  ggplot(aes(x = NMDS1, y = NMDS2)) +
  geom_vline(xintercept = 0, linetype = 'dotted') +
  geom_hline(yintercept = 0, linetype = 'dotted') +
  # Points for samples, coloured by nature
  geom_point(aes(colour = groundwater, shape = medium), size = 3, stroke = 1) +
  theme_tidy() +
  scale_shape_manual(values = c("Culture" = 1, "Environmental" = 19), name = "") +
  scale_color_manual(values = cols.gw, name = "Groundwater") +
  annotate('text', x = -Inf, y = -Inf, size = 7/.pt, 
           label = paste('Stress = ', round(nmds$stress, digits = 2)),
           hjust = -.1, vjust = -1,
           ) +
  ggtitle("0.1 - 0.45 µm fraction") +
  theme(plot.title = element_text(hjust = .5, size = 7),
        axis.title.y = element_blank())
```


```{r plot nmds patchwork, fig.width = 14/2.54}
library(patchwork)
p1 + p2 + plot_layout(guides = "collect") & 
  theme(
    legend.position = "right",
    plot.background = element_blank())
```


```{r export patchwork ordination}
ggsave(filename = "figures/nmds.pdf", width = 16, height = 10, units = "cm")
```


### Microscopy ###

```{r load growth data}
growth <- read_tsv("data/micro_cultures.tsv", col_types = cols(.default = col_character(), 
                                                  count = col_integer(),
                                                  date = col_date(format = "%d-%m-%Y"))
                   ) %>%
  mutate(week = case_when(
    date == "2021-10-15" ~  1,
    date == "2021-11-11" ~  5,
    date == "2021-11-25" ~  7,
    date == "2021-12-09" ~  9,
    date == "2021-12-30" ~ 12,
    date == "2022-01-18" ~ 15,
    date == "2022-02-03" ~ 17
  )) %>%
  filter(!sample %in% c("P26201_1051","P26201_1024")) %>%
  mutate(label = case_when(
    groundwater == "KR0015" ~  "a: meteoric",
    groundwater == "SA1420" ~  "b: marine",
    groundwater == "SA2600" ~  "c: saline"
  ))
```

```{r}
# Microscopy counts on KR0015 groundwater in duplicates, obtaining 680,220 and 890,910 cells ml-1.
# Microscopy counts on SA1420 groundwater in duplicates, obtaining 59220 and 73910 cells ml-1.
# For SA2600, this number was 14,100 and 23,000 cells ml-1

p1 <- growth %>% filter(groundwater == "KR0015") %>%
  ggplot(aes(x = week, y = count, colour = fraction, label = label)) +
  geom_text(aes(x = -Inf, y = Inf), 
            size = 7/.pt, hjust = -0.2, vjust = 2, colour = "black",
            check_overlap = TRUE) +
  geom_point() +
  geom_line(aes(linetype = medium)) +
  geom_hline(yintercept = 785565, linewidth = 0.3) +
  annotate("ribbon", x = c(-Inf, Inf), ymin = 636585, ymax = 934545, alpha = 0.4) +
  # Add sd around the mean
  scale_y_log10(limits = c(1e4,1e8),
                breaks = c(1e4,1e5,1e6,1e7),
                labels = c(expression("10"^"4"),
                           expression("10"^"5"),
                           expression("10"^"6"),
                           expression("10"^"7"))
                ) +
  scale_x_continuous(breaks = c(1,3,5,7,9,11,13,15,17)) +
  scale_colour_manual(values = c("unfiltered" = "#1B5331", "filtered" = "#A2A475"),
                      labels = c("0.1 - 0.45 µm","All cells")) +
  scale_linetype(name = "Medium:", labels = c("SRB","Cell lysate")) +
  labs(x = "", y = expression("Cells (ml"^"-1"*")"), colour = "Fraction") +
  theme_tidy(legend = "bottom") + 
  theme(
    panel.grid.major.y = element_line(colour = "grey", linetype = "dotted", linewidth = 0.3),
    plot.margin = margin(),
    axis.ticks.y = element_blank(),
    panel.border = element_blank(),
    axis.line = element_line(linewidth = 0.4, colour = "black")
    )
```


```{r}
# Microscopy counts on SA1420 groundwater in duplicates, obtaining 59220 and 73910 cells ml-1.

p2 <- growth %>% filter(groundwater == "SA1420") %>%
  ggplot(aes(x = week, y = count, colour = fraction, label = label)) +
  geom_text(aes(x = -Inf, y = Inf), 
            size = 7/.pt, hjust = -0.2, vjust = 2, colour = "black",
            check_overlap = TRUE) +
  geom_point() +
  geom_line(aes(linetype = medium)) +
  geom_hline(yintercept = 66565, linewidth = 0.3) +
  annotate("ribbon", x = c(-Inf, Inf), ymin = 56178, ymax = 76952, alpha = 0.4) +
  # Add sd around the mean
  scale_y_log10(limits = c(1e4,1e8),
                breaks = c(1e4,1e5,1e6,1e7),
                labels = c(expression("10"^"4"),
                           expression("10"^"5"),
                           expression("10"^"6"),
                           expression("10"^"7"))
                ) +
  scale_x_continuous(breaks = c(1,3,5,7,9,11,13,15,17)) +
  scale_colour_manual(values = c("unfiltered" = "#1B5331", "filtered" = "#A2A475"),
                      labels = c("0.1 - 0.45 µm","All cells"),
                      guide = "none") +
  scale_linetype(name = "Medium:", labels = c("SRB","Cell lysate"), guide = "none") +
  labs(x = "Week", y = "") +
  theme_tidy(legend = "none") + 
  theme(
    panel.grid.major.y = element_line(colour = "grey", linetype = "dotted", linewidth = 0.3),
    plot.margin = margin(),
    axis.ticks.y = element_blank(),
    axis.text.y = element_blank(),
    panel.border = element_blank(),
    axis.line = element_line(linewidth = 0.4, colour = "black")
    )
```


```{r}
# For SA2600, this number was 14,100 and 23,000 cells ml-1

p3 <- growth %>% filter(groundwater == "SA2600") %>%
  ggplot(aes(x = week, y = count, colour = fraction, label = label)) +
  geom_text(aes(x = -Inf, y = Inf), 
            size = 7/.pt, hjust = -0.2, vjust = 2, colour = "black",
            check_overlap = TRUE) +
  geom_point() +
  geom_line(aes(linetype = medium)) +
  geom_hline(yintercept = 18550, linewidth = 0.3) +
  annotate("ribbon", x = c(-Inf, Inf), ymin = 12257, ymax = 24843, alpha = 0.4) +
  # Add sd around the mean
  scale_y_log10(limits = c(1e4,1e8),
                breaks = c(1e4,1e5,1e6,1e7),
                labels = c(expression("10"^"4"),
                           expression("10"^"5"),
                           expression("10"^"6"),
                           expression("10"^"7"))
                ) +
  scale_x_continuous(breaks = c(1,3,5,7,9,11,13,15,17)) +
  scale_colour_manual(values = c("unfiltered" = "#1B5331", "filtered" = "#A2A475"),
                      labels = c("0.1 - 0.45 µm","All cells"),
                      guide = "none") +
  scale_linetype(name = "Medium:", labels = c("SRB","Cell lysate"), guide = "none") +
  labs(x = "Week", y = "") +
  theme_tidy(legend = "none") + 
  theme(
    panel.grid.major.y = element_line(colour = "grey", linetype = "dotted", linewidth = 0.3),
    plot.margin = margin(),
    axis.ticks.y = element_blank(),
    axis.text.y = element_blank(),
    panel.border = element_blank(),
    axis.line = element_line(linewidth = 0.4, colour = "black")
    )
```


```{r}
library(patchwork)
p1 + p2 + p3 + plot_layout(guides = "collect") & 
  theme(
    legend.position = "bottom",
    plot.background = element_blank())
```


```{r export growth}
ggsave(filename = "figures/growth.pdf", width = 18, height = 8, units = "cm")
ggsave(filename = "figures/growth.png", width = 18, height = 8, units = "cm")
```

```{r}
# Microscopy counts on KR0015 groundwater in duplicates, obtaining 680,220 and 890,910 cells ml-1.
# Microscopy counts on SA1420 groundwater in duplicates, obtaining 59220 and 73910 cells ml-1.
# For SA2600, this number was 14,100 and 23,000 cells ml-1
i <- data.frame(groundwater = c("KR0015","KR0015","SA1420","SA1420","SA2600","SA2600"), 
                count = c(680220,890910,59220,73910,14100,23000))

i %>%
  mutate(count = count / 1000) %>%
  group_by(groundwater) %>% summarise(sd = sd(count), count = mean(count)) %>%
  ggplot(aes(groundwater, count)) +
  geom_errorbar(aes(ymin = count - sd, ymax = count + sd), width = 0.2, color = "black") +
  geom_col(color = "black", fill = "#BDBDBD") +
  theme_tidy() +
  labs(x = "", y = expression("Cells (" %*%"1000 ml"^"-1"*")")) +
  theme(
    panel.border = element_blank(),
    axis.line = element_line(color = "black")
  )
```
```{r export growth}
ggsave(filename = "figures/cellst0.pdf", width = 8, height = 8, units = "cm")
ggsave(filename = "figures/cellst0.png", width = 8, height = 8, units = "cm")
```



### QPCR KR0015 ###


```{r data qpcr}
read_tsv('data/qpcr.tsv', col_types = cols(.default = col_character(), 
                                      quantity = col_double(),
                                      age = col_factor(levels = seq(1:10) %>% as.character())
                                      )
         ) %>% mutate(medium = gsub("ace","srm", medium)) -> qpcr
```


```{r plot qpcr, fig.width = 12/2.54}
qpcr %>%
  group_by(sample, Kingdom) %>% 
  summarise(mean = mean(quantity), sd = sd(quantity), .groups = 'drop') %>%
  inner_join(smd, by = "sample") %>%
  mutate(category = paste(medium, Kingdom)) %>%
  # Call plot
  ggplot(aes(x = fct_relevel(age, seq(1:10) %>% as.character), 
             y = mean, 
             group = category, 
             colour = fct_relevel(Kingdom, c("Bacteria","Archaea"))
             )
         ) +
  geom_line(aes(linetype = medium)) +
  #facet_wrap(~ fraction) +
  facet_wrap(~ fraction, labeller = as_labeller(c("filtered" = "0.1 - 0.45 µm fraction", "unfiltered" = "All cells"))) +
  geom_pointrange(aes(ymin = mean - sd, 
                      ymax = mean + sd), 
                  size = 0.2) +
  labs(x = 'Week', y = expression("Gene copy number (ml"^"-1"*")")) +
  scale_y_log10(breaks = c(1e3,1e4,1e5,1e6,1e7), 
                labels = c(expression("10"^"3"),
                           expression("10"^"4"),
                           expression("10"^"5"),
                           expression("10"^"6"),
                           expression("10"^"7")
                           )
                ) +
  scale_linetype(name = "Medium:", labels = c("SRB","Cell lysate")) +
  scale_color_manual(name = "", values = c("Bacteria" = "#1B5331","Archaea" = "#A2A475")) +
  theme_tidy(legend = "bottom") +
  theme(panel.grid.major.y = element_line(colour = "grey", linetype = "dotted", linewidth = 0.5),
        axis.ticks.y = element_blank(),
        plot.margin = margin(),
        legend.box.margin = margin(),
        strip.background = element_blank())
```


```{r export qpcr}
ggsave(filename = "figures/qpcr.pdf", width = 12, height = 8, units = "cm")
ggsave(filename = "figures/qpcr.png", width = 12, height = 8, units = "cm")
```


### Areaplot KR0015 ###

```{r select most abundant asvs, message=FALSE, warning=FALSE}
seqtab %>%
  filter(sample %in% paste("P21015", seq(1042,1081), sep = "_")
           ) %>%
  inner_join(tax, by = "seqid") %>%
  group_by(taxon, sample) %>%
  # Sum the abundance of each phylum within a sample
  summarise(relab = sum(relab), .groups = 'drop_last') %>%
  # Calculate the mean abundance of each phylum over the categories
  summarise(mean_relab = sum(relab), .groups = 'drop') %>%
  filter(!is.na(taxon)) %>%
  top_n(9, mean_relab) -> t
```


```{r add selection to taxonomy table, message=FALSE, warning=FALSE}
tax %>%
  left_join(t %>% transmute(taxon, topphylum = taxon), by = "taxon") %>%
  replace_na(list("topphylum" = "Other")) -> taxref
```


```{r areaplot P21015 phylum}
seqtab %>%
  filter(sample %in% paste("P21015", seq(1042,1081), sep = "_")
           ) %>%
  inner_join(taxref, by = "seqid") %>%
  # Summarize in order to have the sum for each category and topphylum
  group_by(sample, topphylum) %>% 
  summarise(relab = sum(relab), .groups = 'drop') %>%
  inner_join(smd, by = "sample") %>%
  mutate(fraction = if_else(fraction == 'filtered', 
                       '0.1 - 0.45 µm fraction', 
                       '> 0.1 µm fraction')
         ) %>%
  mutate(medium = if_else(medium == 'lys', 
                       'Cell lysate', 
                       'SRB')
         ) %>%
  mutate(fraction = paste(medium, fraction)) %>%
  mutate(age = as.numeric(.$age)) %>% 
  # Call the plot
  ggplot(aes(x = age, 
             y = relab, 
             fill = fct_relevel(topphylum, c("Other")))
         ) +
  geom_area() + scale_x_continuous(n.breaks = 10) +
  facet_wrap(~ fraction) +
  labs(x = 'Week', y = 'Relative abundance') +
  scale_fill_brewer(palette = 'Paired') +
  theme_tidy(fontsize = 7) +
  theme(
    legend.position = "right",
    legend.key.size = unit(4, "mm"),
    strip.text = element_text(size = 7, face = "bold"))
```


```{r export area plot}
ggsave("figures/areaplot_phylum.pdf", height = 14, width = 16, units = "cm")
```

### Overview community structure t0 ###


```{r most abundant phyla (t0)}
# Use the samples from P15009 and P21015 as t0
t0 <- c(
  "P15009_1057","P15009_1059","P15009_1060","P21015_1020","P21015_1021","P21015_1022", # KR0015
  "P15009_1061","P15009_1065","P15009_1066","P21015_1023","P21015_1024","P21015_1025", # SA1420
  "P15009_1069","P15009_1062","P15009_1063","P21015_1029","P21015_1030","P21015_1031"  # SA2600
       )

seqtab %>%
  filter(sample %in% t0) %>%
  inner_join(tax, by = "seqid") %>%
  group_by(phylum, sample) %>%
  # Sum the abundance of each phylum within a sample
  summarise(relab = sum(relab), .groups = 'drop_last') %>%
  # Calculate the mean abundance of each phylum over the categories
  summarise(mean_relab = sum(relab), .groups = 'drop') %>%
  filter(!is.na(phylum)) %>%
  top_n(8, mean_relab) -> t

tax %>%
  left_join(t %>% transmute(phylum, topphylum = phylum), by = "phylum") %>%
  mutate(topphylum = if_else(is.na(phylum), "Unidentified", topphylum)) %>%
  replace_na(list("topphylum" = "Other")) -> taxref
```


```{r plot environmental communities (t0)}
seqtab %>%
  filter(sample %in% t0) %>%
  inner_join(taxref, by = "seqid") %>%
  # Summarize in order to have the sum for each category and topphylum
  group_by(sample, topphylum) %>% 
  summarise(mrelab = sum(relab), .groups = 'drop') %>%
  # Call the plot
  ggplot(aes(x = fct_relevel(sample, t0), 
             y = mrelab * 100, 
             fill = fct_relevel(topphylum, names(cols.phylum))
             )
         ) +
  labs(x = '', y = 'Relative abundance (%)', fill = "Phylum") +
  geom_col() + scale_y_continuous(expand = c(0.01,0), limits = c(-9,101)) +
  annotate(geom = "segment", x =  0.55, xend =  3.45, y = -1, yend = -1) +
  annotate(geom = "segment", x =  3.55, xend =  6.45, y = -1, yend = -1) +
  annotate(geom = "segment", x =  6.55, xend =  9.45, y = -1, yend = -1) +
  annotate(geom = "segment", x =  9.55, xend = 12.45, y = -1, yend = -1) +
  annotate(geom = "segment", x = 12.55, xend = 15.45, y = -1, yend = -1) +
  annotate(geom = "segment", x = 15.55, xend = 18.45, y = -1, yend = -1) +
  # Labels dates
  annotate(geom = "text", x =  2, y = -3.5, label = "Oct 2019", size = 7/.pt) +
  annotate(geom = "text", x =  5, y = -3.5, label = "April 2020", size = 7/.pt) +
  annotate(geom = "text", x =  8, y = -3.5, label = "Oct 2019", size = 7/.pt) +
  annotate(geom = "text", x = 11, y = -3.5, label = "April 2020", size = 7/.pt) +
  annotate(geom = "text", x = 14, y = -3.5, label = "Oct 2019", size = 7/.pt) +
  annotate(geom = "text", x = 17, y = -3.5, label = "April 2020", size = 7/.pt) +
  # Labels groundwater
  annotate(geom = "text", x = 3.5, y = -8, label = "KR0015", size = 7/.pt) +
  annotate(geom = "text", x = 9.5, y = -8, label = "SA1420", size = 7/.pt) +
  annotate(geom = "text", x =15.5, y = -8, label = "SA2600", size = 7/.pt) +
  scale_fill_manual(values = cols.phylum) +
  theme_tidy(ratio = 0.7) + 
  theme(axis.text.x = element_blank(), 
        axis.ticks.x = element_blank(),
        legend.key.size = unit(3.5, "mm"),
        legend.margin = margin()
        )
```


```{r export t0 plot}
ggsave("figures/t0_phylum.pdf", height = 8, width = 14, units = "cm")
ggsave("figures/t0_phylum.png", height = 8, width = 14, units = "cm")

```


```{r}
i <- seqtab %>% inner_join(tax, by = "seqid") %>% 
  filter(phylum == "Desulfobacterota") %>% 
  group_by(sample, phylum) %>% summarise(relab = sum(relab), .groups = "drop") %>% 
  filter(!sample %in% t0) %>% # Use only cultures
  slice_max(relab, n = 20) %>% pull(sample) # Select the 20 most abundant cultures

i <- adiv %>% slice_min(shannon, n = 20) %>% pull(sample)

seqtab %>%
  filter(sample %in% i) %>%
  inner_join(tax, by = "seqid") %>%
  group_by(phylum, sample) %>%
  # Sum the abundance of each phylum within a sample
  summarise(relab = sum(relab), .groups = 'drop_last') %>%
  # Calculate the mean abundance of each phylum over the categories
  summarise(mean_relab = sum(relab), .groups = 'drop') %>%
  filter(!is.na(phylum)) %>%
  top_n(6, mean_relab) %>% arrange(desc(mean_relab)) -> t

tax %>%
  left_join(t %>% transmute(phylum, topphylum = phylum), by = "phylum") %>%
  #mutate(topphylum = if_else(is.na(phylum), "Unidentified", topphylum)) %>% # Most is identified in cultures so turn off
  replace_na(list("topphylum" = "Other")) -> taxref

seqtab %>%
  filter(sample %in% i) %>%
  inner_join(taxref, by = "seqid") %>%
  # Summarize in order to have the sum for each category and topphylum
  group_by(sample, topphylum) %>% 
  summarise(mrelab = sum(relab), .groups = 'drop') %>%
  # Call the plot
  ggplot(aes(sample, y = mrelab * 100, 
             fill = fct_relevel(topphylum, t$phylum)
             )
         ) +
  labs(x = '', y = 'Relative abundance (%)', fill = "Phylum") +
  geom_col() + scale_y_continuous(expand = c(0.01,0)) +
  scale_fill_manual(values = c("Patescibacteria" = "#C7B19C","Nanoarchaeota" = "#046C9A", "Firmicutes" = "#1B5331", "Bacteroidota" = "#046C9A", "Campylobacterota" = "#FAEFD1",
                                "Omnitrophota" = "#00A08A", "Proteobacteria" = "#D69C4E", "Desulfobacterota" = "#A2A475",
                                "Other" = "#899DA4", "Unidentified" = "#046C9A")) +
  theme_tidy(ratio = 0.7) + 
  theme(axis.text.x = element_blank(), 
        axis.ticks.x = element_blank(),
        legend.key.size = unit(3.5, "mm"),
        legend.margin = margin()
        )
```


```{r export t0 plot}
ggsave("figures/patesci.pdf", height = 8, width = 14, units = "cm")
ggsave("figures/patesci.png", height = 8, width = 14, units = "cm")

ggsave("figures/monoculture.pdf", height = 8, width = 14, units = "cm")
ggsave("figures/monoculture.png", height = 8, width = 14, units = "cm")
```


```{r}
i <- c("P21015_1043","P21015_1047","P21015_1051","P21015_1055","P21015_1059", # SRB
       "P21015_1063","P21015_1067","P21015_1071","P21015_1075","P21015_1079", # SRB
       "P21015_1045","P21015_1049","P21015_1053","P21015_1057","P21015_1061", # Lysate
       "P21015_1065","P21015_1069","P21015_1073","P21015_1077","P21015_1081") # Lysate

seqtab %>%
  filter(sample %in% i) %>%
  inner_join(tax, by = "seqid") %>%
  group_by(phylum, sample) %>%
  # Sum the abundance of each phylum within a sample
  summarise(relab = sum(relab), .groups = 'drop_last') %>%
  # Calculate the mean abundance of each phylum over the categories
  summarise(mean_relab = sum(relab), .groups = 'drop') %>%
  filter(!is.na(phylum)) %>%
  top_n(4, mean_relab) %>% arrange(desc(mean_relab)) -> t

tax %>%
  left_join(t %>% transmute(phylum, topphylum = phylum), by = "phylum") %>%
  #mutate(topphylum = if_else(is.na(phylum), "Unidentified", topphylum)) %>% # Most is identified in cultures so turn off
  replace_na(list("topphylum" = "Other")) -> taxref

seqtab %>%
  filter(sample %in% i) %>%
  inner_join(taxref, by = "seqid") %>%
  # Summarize in order to have the sum for each category and topphylum
  group_by(sample, topphylum) %>% 
  summarise(mrelab = sum(relab), .groups = 'drop') %>%
  # Call the plot
  ggplot(aes(fct_relevel(sample,i), y = mrelab * 100, 
             fill = fct_relevel(topphylum, t$phylum)
             )
         ) +
  labs(x = '', y = 'Relative abundance (%)', fill = "Phylum") +
  geom_col() + scale_y_continuous(expand = c(0.01,0),limits = c(-4,101)) +
  scale_fill_manual(values = c("Patescibacteria" = "#C7B19C","Nanoarchaeota" = "#046C9A", 
                                "Spirochaetota" = "#00A08A", "Proteobacteria" = "#D69C4E", "Desulfobacterota" = "#A2A475",
                                "Other" = "#899DA4", "Acidobacteriota" = "#046C9A")) +
  theme_tidy(ratio = 0.7) + 
  annotate(geom = "segment", x =  0.55, xend =  10.45, y = -1, yend = -1) +
  annotate(geom = "segment", x = 10.55, xend =  20.45, y = -1, yend = -1) +
  annotate(geom = "text", x = 5.5, y = -3, label = "SRB", size = 7/.pt) +
  annotate(geom = "text", x =15.5, y = -3, label = "Lysate", size = 7/.pt) +
  theme(axis.text.x = element_blank(), 
        axis.ticks.x = element_blank(),
        legend.key.size = unit(3.5, "mm"),
        legend.margin = margin()
        )
```
```{r export t0 plot}
ggsave("figures/baproject_filtered.pdf", height = 8, width = 14, units = "cm")
ggsave("figures/baproject_filtered.png", height = 8, width = 14, units = "cm")
```


```{r most abundant taxa}
seqtab %>% 
  inner_join(smd) %>% 
  filter(groundwater %in% c("SA1420","KR0015","SA2600") & medium == "env") %>% 
  inner_join(tax) %>% group_by(taxon) %>% 
  summarise(n = sum(relab)) %>% 
  slice_max(n, n = 25) %>%
  pull(taxon) -> t

seqtab %>%
  inner_join(tax, by = "seqid") %>% filter(taxon %in% t) %>%
  group_by(sample, taxon) %>% summarise(relab = sum(relab)*100, .groups = "drop") %>%
  inner_join(smd, by = "sample") %>% 
  filter(groundwater %in% c("SA1420","KR0015","SA2600")) %>%
  filter(relab >= 1) %>%
  ggplot(aes(x = relab, y = fct_relevel(taxon, rev(t)))
         ) +
  geom_boxplot(fill = "#d9d9d9", size = 0.3) +
  scale_x_log10(breaks = c(0.1,1,10,100), labels = c("0.1", "1", "10", "100")) +
  labs(x = "Relative abundance (%)") +
  facet_wrap(~groundwater) +
  theme_tidy(fontsize = 7, ratio = 2) +
  theme(
    axis.title.y = element_blank(),
    axis.ticks.y = element_blank(),
    axis.text.y = element_text(face = "plain"),
    strip.text = element_text(size = 7, face = "bold")
  )
```


```{r export boxplot abundant taxa}
ggsave("figures/boxplot_top.pdf", height = 10, width = 16, units = "cm")
```


### Redundancy analysis ###


```{r rda-metadata}
# t0 are the environmental samples with n = 6 for each groundwater type
t0 <- c(
  "P15009_1057","P15009_1059","P15009_1060","P21015_1020","P21015_1021","P21015_1022", # KR0015
  "P15009_1061","P15009_1065","P15009_1066","P21015_1023","P21015_1024","P21015_1025", # SA1420
  "P15009_1069","P15009_1062","P15009_1063","P21015_1029","P21015_1030","P21015_1031"  # SA2600
       )

geochemistry <- list( # Concentrations are in mg l-1
  groundwater = c("KR0015","SA1420","SA2600"),
  strontium = c(0.716,0.716,0.719),
  sulfate = c(60.3,314,630),
  ph = c(7.4,7.5,7.5),
  ec = c(264,958,3550),
  doc = c(15.4,6.6,1.0),
  oxygen = c(-10.5,-7.45,-12.2),
  nitrogen = c(0.25,1.82,0.08),
  sulfur = c(0.482,0.094,0),
  s34 = c(27.4,30.0,13.1)
  ) %>% data.frame() %>%
  inner_join(smd[,c("sample","groundwater")], by = "groundwater")

geochemistry <- seqtab %>% # Add the species richness for each sample
  group_by(sample) %>% add_tally() %>% ungroup() %>%
  select(sample, n) %>% distinct() %>% inner_join(geochemistry, by = "sample")
```


```{r rda-data}
hellinger <- seqtab %>%
  filter(sample %in% t0) %>%
  # Standard Vegan transformation: Turn table with with samples as *rows*
  select(sample, seqid, count) %>%
  spread(seqid, count, fill = 0) %>%
  # Turn into a numeric matrix
  column_to_rownames('sample') %>% vegan::decostand(method = "hellinger") %>%
  rownames_to_column("sample") %>%
  pivot_longer(-1, names_to = "seqid", values_to = "hellinger")

m <- hellinger %>% # Create a matrix
  spread(seqid, hellinger) %>%
  column_to_rownames("sample")

hellinger <- seqtab %>%
  inner_join(tax, by = "seqid") %>%
  filter(sample %in% t0) %>%
  group_by(phylum, sample) %>% summarise(count = sum(count), .groups = "drop") %>%
  # Standard Vegan transformation: Turn table with with samples as *rows*
  select(sample, phylum, count) %>%
  spread(phylum, count, fill = 0) %>%
  # Turn into a numeric matrix
  column_to_rownames('sample') %>% vegan::decostand(method = "hellinger") %>%
  rownames_to_column("sample") %>%
  pivot_longer(-1, names_to = "phylum", values_to = "hellinger")

m <- hellinger %>% # Create a matrix
  spread(phylum, hellinger) %>%
  column_to_rownames("sample")
```


```{r rda}
vegan::rda(
  m ~ oxygen + doc + s34, correlation = F,
  data = geochemistry %>%
    semi_join(hellinger, by = 'sample') %>%
    filter(sample %in% t0) %>%
    arrange(sample) %>%
    column_to_rownames('sample') 
) -> rda

# This one will contain the proportions explained which we get by dividing the
# eigenvalue of each component with the sum of eigenvalues for all components.
p <- c(rda$CCA$eig, rda$CA$eig)/sum(c(rda$CCA$eig, rda$CA$eig))
# This one is collecting the coordinates for arrows that will depict how the 
# factors in our model point in the coordinate
a <- rda$CCA$biplot %>% data.frame() %>% tibble::rownames_to_column('factor')

lab <- c("{}^34*S~'in'~SO[4]^{2-phantom()}", "delta^18*O", "Dissolved~organic~carbon") %>% cbind(a) %>% rename(label = 1) %>% select(-factor)
# sulfate (RDA1, RDA2) = (0.95,0.31)
# ec (1.0, 0.05)
# nitrogen (-0.45, 0.89)
# s34 (-0.95,0.28)
a <- list(factor = c("s34"),
          RDA1 = c(-0.96),
          RDA2 = c( 0.29)) %>% data.frame() %>% rbind(a)
```


```{r rda-plot, fig.width = 10/2.53}
rda$CCA$wa %>% data.frame() %>% tibble::rownames_to_column('sample') %>%
  inner_join(geochemistry, by = 'sample') %>%
  ggplot(aes(x = RDA1, y = RDA2)) +
  geom_vline(xintercept = 0, colour = "grey", linetype = "dotted") +
  geom_hline(yintercept = 0, colour = "grey", linetype = "dotted") +
  geom_point(aes(colour = groundwater), shape = 1, stroke = 1) +
  xlab(sprintf("RDA1 (%2.1f%% explained)", p[[1]] * 100)) +
  ylab(sprintf("RDA2 (%2.1f%% explained)", p[[2]] * 100)) +
  geom_segment(
    data = a, mapping = aes(xend = RDA1/4, yend = RDA2/4), 
    x = 0, y = 0, arrow = arrow(length = unit(2, "mm"), type = "closed") 
    ) +
  geom_text(
    data = a, mapping = aes(x = RDA1/8, y = RDA2/8, label = lab$label),
    size = 7/.pt, parse = TRUE
    ) + 
  scale_size(range = c(1,8), name = "") +
  annotate("text", x = -0.1, y =-0.25, label = "KR0015", colour = cols.gw[1], size = 7/.pt) +
  annotate("text", x = 0.05, y = 0.28, label = "SA1420", colour = cols.gw[2], size = 7/.pt) +
  annotate("text", x =  0.2, y =-0.05, label = "SA2600", colour = cols.gw[3], size = 7/.pt) +
  stat_ellipse(aes(x = RDA1, y = RDA2, colour = groundwater), 
               geom = "polygon", show.legend = F,
               fill = NA, linetype = "dashed") +
  scale_color_manual(name = "",values = cols.gw, guide = "none") +
  scale_fill_manual(name = "",values = cols.gw, guide = "none") +
  theme_tidy(legend = "bottom")
```
```{r export rda}
ggsave("figures/rdax.pdf", height = 12, width = 10, units = "cm")
```


```{r}
chemistry <- read_tsv("../Amplicons/chemistry.txt", col_types = cols(.default = col_double(), replicate = col_character())) %>%
  mutate(Cl = Cl / 1000, EC = EC / 100) %>%
  gather(parameter, value, -1) %>% 
  na.omit() %>% filter(!grepl("SSM", replicate))

chemistry %>% mutate(dummy = "dummy") %>%
  filter(parameter %in% c("depth","O18","SO4","DOC","Sr")) %>%
  ggplot(aes(dummy, value)) + 
  geom_boxplot() + labs(x = "", y = "") + theme_tidy(ratio = 2) +
  facet_wrap(~ parameter, nrow = 1, scales = "free", labeller = 
               as_labeller(c(
                 depth = "Depth~(m)", 
                 DOC = "Org.~C~(mg~L^-1)",
                 O18 = 'delta^{18}*O~(ppt)', 
                 SO4 = 'SO[4]^{2-phantom()} ~ (mg~L^-1)',
                 Sr = 'Sr ~ (mg~L^-1)'), default = label_parsed)
             ) + theme(
               axis.text.x = element_blank(),
               axis.ticks.x = element_blank(),
               strip.background = element_blank(),
               panel.border = element_blank(),
               axis.line = element_line(color = "black")
             )
```


```{r}
ggsave("chemistry.png", width = 14, height = 5, units = "cm")
```


### Rarefaction plus sampling site ###


```{r rarefaction}
seqtab %>%
  inner_join(smd, by = "sample") %>% filter(groundwater == "KR0015B") %>%
  select(sample, seqid, count) %>%
  spread(seqid, count, fill = 0) %>%
  column_to_rownames('sample') -> seqtab_matrix
  
png(filename = "figures/rarecurve_meteoric.png", width = 14, height = 10, units = "cm", res = 300)  
min(rowSums(seqtab_matrix)) %>%
vegan::rarecurve(seqtab_matrix, 
                 sample = ., step = 100, cex = 0.5, cex.axis = 0.6, cex.lab = 0.6, 
                 xlab = 'Sequencing depth KR0015B', 
                 ylab = 'Rarefied No. of ASVs', label = TRUE)
dev.off()


seqtab %>%
  inner_join(smd, by = "sample") %>% filter(groundwater == "SA1420A-1") %>%
  select(sample, seqid, count) %>%
  spread(seqid, count, fill = 0) %>%
  column_to_rownames('sample') -> seqtab_matrix
  
png(filename = "figures/rarecurve_mm.png", width = 14, height = 10, units = "cm", res = 300)  
min(rowSums(seqtab_matrix)) %>%
vegan::rarecurve(seqtab_matrix, 
                 sample = ., step = 100, cex = 0.5, cex.axis = 0.6, cex.lab = 0.6, 
                 xlab = 'Sequencing depth SA1420A-1', 
                 ylab = 'Rarefied No. of ASVs', label = TRUE)
dev.off()


seqtab %>%
  inner_join(smd, by = "sample") %>% 
  filter(groundwater %in% c("KA3511A-1","SA2600A-1","KA2862A-1")) %>%
  select(sample, seqid, count) %>%
  spread(seqid, count, fill = 0) %>%
  column_to_rownames('sample') -> seqtab_matrix
  
png(filename = "figures/rarecurve_os.png", width = 14, height = 10, units = "cm", res = 300)  
min(rowSums(seqtab_matrix)) %>%
vegan::rarecurve(seqtab_matrix, 
                 sample = ., step = 100, cex = 0.5, cex.axis = 0.6, cex.lab = 0.6, 
                 xlab = 'Sequencing depth SA2600, KA3511 & KA2862', 
                 ylab = 'Rarefied No. of ASVs', label = TRUE)
dev.off()
```


```{r}
# Äspö Lat N 57.4334, Lon E 16.6603) 
baltic <- list.files("data/shapefiles/", pattern = ".shp", full.names = TRUE) %>% lapply(sf::st_read)

ggplot() +
  # Surrounding countries with alpha
  geom_sf(data = baltic[[1]], aes(fill = var1), colour = "#f3e2d0", fill = "#f8efe4", alpha = 0.8) +
  geom_sf(data = baltic[[2]], aes(fill = var1), colour = "#f3e2d0", fill = "#f8efe4", alpha = 0.8) +
  geom_sf(data = baltic[[3]], aes(fill = var1), colour = "#f3e2d0", fill = "#f8efe4", alpha = 0.8) +
  geom_sf(data = baltic[[4]], aes(fill = var1), colour = "#f3e2d0", fill = "#f8efe4", alpha = 0.8) +
  geom_sf(data = baltic[[5]], aes(fill = var1), colour = "#f3e2d0", fill = "#f8efe4", alpha = 0.8) +
  geom_sf(data = baltic[[6]], aes(fill = var1), colour = "#f3e2d0", fill = "#f8efe4", alpha = 0.8) +
   # Sweden in full color
  geom_sf(data = baltic[[7]], aes(fill = var1), colour = "#e3bc93", fill = "#f3e2d0") +
  ggspatial::annotation_scale(location = "br", text_cex = 0.5) +
  annotate("point", x = 16.6603, y = 57.4334, 
           size = 3, stroke = 2, colour = "#c72b17", fill = "white", shape = 21
           ) +
  annotate("text", x = 15, y = 59.5, label = "Sweden", colour = "#b59675", size = 7/.pt) +
  annotate("text", x = 18.5, y = 56.0, label = "Baltic Sea", colour = "#6f8b99", size = 7/.pt) +
  lims(x = c(10,23), y = c(55,66)) + labs(x = "", y = "") + theme_tidy(ratio = NULL, fontsize = 6) +
  theme(panel.background = element_rect(fill = "#bac7ce",linewidth = 0), 
        panel.border = element_blank(),
        axis.text.x = element_blank(),
        axis.text.y = element_blank(), 
        axis.ticks = element_blank()
        )


ggplot() + 
  geom_sf(data = filter(baltic[[8]], NAME %in% c("Baltic Sea","Gulf of Bothnia")), 
          aes(fill = var1), colour = "#637e8b", fill = "#8da3ae") +  
  geom_sf(data = baltic[[7]], colour = "#e3bc93", fill = "#f3e2d0") +

  ggspatial::annotation_scale(location = "br") +
  annotate("point", x = 16.6603, y = 57.4334, 
           size = 3, stroke = 2, colour = "#c72b17", fill = "white", shape = 21
           ) +
  annotate("text", x = 15, y = 59.5, label = "Sweden", colour = "#b59675", size = 7/.pt) +
  annotate("text", x = 18.5, y = 56.0, label = "Baltic Sea", colour = "#6f8b99", size = 7/.pt) +
  theme_void()
```

```{r export se-sweden}
ggsave("figures/se-sweden-void.png", height = 10, width = 6, units = "cm")
```


### Stats metagenomes ###


```{r prepare checkm data}
t <- c("Firmicutes","Desulfobacterota","Proteobacteria","Patescibacteria","Nanoarchaeota","Other")

# Prepare data for plot
i <- checkm %>%
  inner_join(bintax, by = "genome") %>%
  mutate(phylum = ifelse(phylum %in% t, phylum, "Other")) %>%
  mutate(phylum = factor(phylum, levels = t)) %>%
  select(genome, phylum, completeness, GC, genome.size, coding.density) %>%
  distinct() %>%
  mutate(genome.size = genome.size/1e6)
```


```{r plot bin stats, fig.width = 10/2.53}
p1<- i %>%
  ggplot(aes(genome.size, GC, color = phylum)) +
  geom_smooth(method = "lm", formula = y ~ x, colour = "black", fill = "#d9d9d9", se = TRUE, linewidth = 0.4) +
  geom_point(size = 2.5, shape = 1, stroke = 1.25) + 
  labs(x = "Estimated genome size (Mbp)", y = "GC content (%)", color = "") +
  scale_colour_manual(values = cols.mag) +
  theme_tidy() + scale_x_continuous(breaks = c(1,2,3,4,5,6)) +
  annotate("text", x = Inf, y = -Inf, 
           label = "paste(R ^ 2, \" = 0.43\")", 
           size = 7/.pt, vjust = -0.5, hjust = 1, parse = TRUE) +
  theme(
    axis.line = element_line(linewidth = 0.4, color = "black"),
    panel.border = element_blank()
  )
```


```{r, fig.width = 2.54}
p2 <- i %>%
  ggplot(aes(phylum, completeness, group = phylum, 
             fill = phylum, color = phylum)
         ) +
  geom_boxplot(alpha = 0.5, key_glyph = "rect") +
  labs(x = "", y = "Completeness (%)", fill = "") +
  scale_fill_manual(values = cols.mag, guide = "none") + 
  scale_color_manual(values = cols.mag, guide = "none") +
  theme_tidy(legend = "bottom", ratio = 1.4) +
  theme(
    axis.line = element_line(linewidth = 0.4, "black"),
    panel.border = element_blank(),
    axis.text.x = element_blank(),
    axis.ticks.x = element_blank(),
    axis.title.y = element_text(margin = margin()),
    plot.margin = margin()
  )
```


```{r, fig.width = 8/2.54}
p3 <- i %>%
  ggplot(aes(genome.size, coding.density, color = phylum)) +
  geom_smooth(method = "lm", formula = y ~ x, colour = "black", fill = "#d9d9d9", se = TRUE, linewidth = 0.4) +
  geom_point(size = 2.5, shape = 1, stroke = 1.25) + 
  labs(x = "Estimated genome size (Mbp)", y = "Coding density", color = "") +
  scale_colour_manual(values = cols.mag, guide = "none") +
  theme_tidy() + scale_x_continuous(breaks = c(1,2,3,4,5,6)) +
  annotate("text", x = Inf, y = Inf, 
           label = "paste(R ^ 2, \" = 0.22\")", 
           size = 7/.pt, vjust = 1, hjust = 1, parse = TRUE) +
  theme(
    axis.line = element_line(linewidth = 0.4, color = "black"),
    panel.border = element_blank()
  )
```


```{r, fig.width = 18/2.54}
library(patchwork)
p1 + p2 + p3 + plot_layout(guides = "collect") &
  theme(legend.position = "bottom",
        axis.ticks.length = unit(".75", "mm"))
```



```{r export growth}
ggsave(filename = "figures/binstats.pdf", width = 18, height = 10, units = "cm")
```
