---
title: "didactic-spork (P26201 - cultures)"
author: "George"
date: '2022-07-12'
output:
  bookdown::html_document2:
    toc: true
    toc_float: true
    toc_depth: 3
    number_sections: true
    theme: spacelab
    highlight: textmate
    df_print: paged
    code_folding: hide
    self_contained: false
    keep_md: false
    encoding: "UTF-8"
---

```{r themes}
source("/Users/geweaa/Box Sync/Protocols/theme.R")
```

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, cache = TRUE, fig.path = "/figures")
library(tidyverse)
theme_set(theme_tidy())
```

# KR0015: P15009_1057, 1059 and 1060 + P21015_1020, 1021 and 1022
# SA1420: P15009_1061, 1065 and 1066 + P21015_1023, 1024 and 1025
# SA2600: P15009_1069, 1062 and 1063 + P21015_1029, 1030 and 1031
# P15009 was sampled Oct 2019, P21015 was sampled Feb 2020, QK-1636 was sampled between Feb and May 2017
# QK-1636 refers to the survey study, KR0015 not included in this study
# QK1636-S58, S59 and S60 for SA1420, S40 + S41 and S42 for SA2600. 


```{r files}
seqtab <- read_tsv("data/ASV_table.tsv", col_types = cols(.default = col_character(), 
                                    count = col_integer()
                                    )) 

# Use GTDB release R07-RS207 throughout the analysis (version 207)
# Cite https://doi.org/10.17044/scilifelab.14869077.v4
# P15009 (Sci-life contract id M.Dopson_19_05) was sampled for cultures and feeding experiment
# P21015 (Sci-life contract id M.Dopson_21_01) was sampled for Larwence Berkeley National Laboratory
# P15009 was sampled Oct 2019, P21015 was sampled Feb 2020
# QK-1636 refers to the survey study, KR0015 not included in this study
# QK1636-S58, S59 and S60 for SA1420, S40 + S41 and S42 for SA2600. 

tax <- read_tsv("data/ASV_tax.tsv", col_types = cols(.default = col_character())
                ) %>% mutate(taxon = coalesce(genus,family,order,class,phylum,kingdom,domain))
smd <- read_tsv("data/smd.tsv", col_types = cols(.default = col_character())
                ) %>% 
  filter(groundwater %in% c("KR0015","SA1420","SA2600")) 

seqtab %>% 
  inner_join(smd, by = "sample") %>% 
  filter(age == "ctrl" & medium == "ace") %>% 
  pull(seqid) %>% unique() -> ace # Contaminant ASVs in acetate medium
seqtab %>% 
  inner_join(smd, by = "sample") %>% 
  filter(age == "ctrl" & medium == "lys") %>% 
  pull(seqid) %>% unique() -> lys # Contaminant ASVs in lysate medium

seqtab %>%
  inner_join(smd, by = "sample") %>%
  filter(
    age != "ctrl" |
    medium == "env" |
    substr(sample,1,6) == "P21015" & medium == "ace" & !seqid %in% ace |
    substr(sample,1,6) == "P21015" & medium == "lys" & !seqid %in% lys |
    substr(sample,1,6) == "P26201"
    ) %>%
  group_by(sample) %>% mutate(relab = count/sum(count)) %>% ungroup() %>%
  select(sample,seqid,count,relab) -> seqtab
```


```{r define colours}
cols.gw <- c("KR0015" = "#b15928",
             "SA1420" = "#1f78b4",
             "SA2600" = "#33a02c")

cols.fraction <- c("Environmental" = "#d9d9d9",
                   "0.1 - 0.45 µm" = "#1f78b4",
                   "> 0.1 µm" = "#a6cee3")
```


### Alpha diversity ###

Estimate alpha diversity using the Shannon-Weaver index on species level

```{r alpha diversity}
seqtab %>%
  select(-relab) %>%
  spread(seqid, count, fill = 0) %>% 
  column_to_rownames('sample') %>%
  vegan::diversity() %>% as.data.frame() %>%
  rownames_to_column('sample') %>%
  rename(shannon = 2) %>%
  inner_join(smd, by = "sample") -> adiv
```


```{r plot alpha diversity, fig.width = 10/2.53}
adiv %>%
  mutate(fraction = if_else(fraction == 'filtered',
                            '0.1 - 0.45 µm',
                            '> 0.1 µm')
         ) %>%
  replace_na(list("fraction" = "Environmental")) %>%
  ggplot(aes(x = fct_relevel(groundwater, c("KR0015","SA1420","SA2600")),
             y = shannon)) +
  geom_boxplot(aes(fill = fct_relevel(fraction, c("Environmental"))), 
               outlier.color = "white") +
  labs(x = "", y = "Alpha diversity (Shannon H index)", fill = "") +
  theme_tidy(legend = c(0.85,0.85)) +
  scale_fill_manual(values = cols.fraction) +
  theme(legend.box.background = element_rect(linetype = "longdash", linewidth = .4, fill = NA), 
        legend.margin = margin(0,2,2,0),
        panel.grid.major.y = element_line(colour = "grey", linetype = "dotted", linewidth = 0.4))
```


```{r export alphadiv}
ggsave(filename = "figures/adiv.pdf", width = 10, height = 10, units = c("cm"))
```


### NMDS cultures ###

```{r nmds unfiltered}
set.seed(999)
nmds <- seqtab %>% # Full size fraction
  inner_join(smd, by = "sample") %>%
  filter(fraction == "unfiltered" | is.na(fraction)) %>%
  group_by(seqid) %>% filter(count > 2) %>% ungroup() %>%
  select(seqid, sample, count) %>% spread(seqid, count, fill = 0) %>%
  column_to_rownames("sample") %>% vegan::metaMDS()

vegan::scores(nmds)$sites %>%
  as.data.frame() %>%
  rownames_to_column("sample") %>%
  inner_join(smd, by = "sample") -> nmds.scores
```

Plot the ordination

```{r plot nmds unfiltered, include = T}
p1 <- nmds.scores %>%
  mutate(medium = if_else(medium == "env","Environmental","Culture")) %>%
  # Plot
  ggplot(aes(x = NMDS1, y = NMDS2)) +
  geom_vline(xintercept = 0, linetype = 'dotted') +
  geom_hline(yintercept = 0, linetype = 'dotted') +
  # Points for samples, coloured by nature
  geom_point(aes(colour = groundwater, shape = medium), size = 3, stroke = 1) +
  theme_tidy() +
  scale_shape_manual(values = c("Culture" = 1, "Environmental" = 19), name = "") +
  scale_color_manual(values = cols.gw, name = "Groundwater") +
  annotate('text', x = -Inf, y = -Inf, size = 7/.pt, 
           label = paste('Stress = ', round(nmds1$stress, digits = 2)),
           hjust = -0.1, vjust = -1,
           ) +
  ggtitle("> 0.1 µm fraction") +
  theme(
    plot.title = element_text(hjust = .5, size = 7)
    )
```


```{r extract nmds coordinates filtered}
nmds <- seqtab %>% # Small size fraction
  inner_join(smd, by = "sample") %>%
  filter(fraction == "filtered" | is.na(fraction)) %>%
  group_by(seqid) %>% filter(count > 2) %>% ungroup() %>%
  select(seqid, sample, count) %>% spread(seqid, count, fill = 0) %>%
  column_to_rownames("sample") %>% vegan::metaMDS()

vegan::scores(nmds)$sites %>%
  as.data.frame() %>%
  rownames_to_column("sample") %>%
  inner_join(smd, by = "sample") -> nmds.scores
```


```{r plot nmds filtered}
p2 <- nmds.scores %>%
  mutate(medium = if_else(medium == "env","Environmental","Culture")) %>%
  # Plot
  ggplot(aes(x = NMDS1, y = NMDS2)) +
  geom_vline(xintercept = 0, linetype = 'dotted') +
  geom_hline(yintercept = 0, linetype = 'dotted') +
  # Points for samples, coloured by nature
  geom_point(aes(colour = groundwater, shape = medium), size = 3, stroke = 1) +
  theme_tidy() +
  scale_shape_manual(values = c("Culture" = 1, "Environmental" = 19), name = "") +
  scale_color_manual(values = cols.gw, name = "Groundwater") +
  annotate('text', x = -Inf, y = -Inf, size = 7/.pt, 
           label = paste('Stress = ', round(nmds$stress, digits = 2)),
           hjust = -.1, vjust = -1,
           ) +
  ggtitle("0.1 - 0.45 µm fraction") +
  theme(plot.title = element_text(hjust = .5, size = 7),
        axis.title.y = element_blank())
```


```{r plot nmds patchwork, fig.width = 16/2.53}
library(patchwork)
p1 + p2 + plot_layout(guides = "collect") & 
  theme(
    legend.position = "right",
    plot.background = element_blank())
```


```{r export patchwork ordination}
ggsave(filename = "figures/nmds.pdf", width = 16, height = 10, units = c("cm"))
```


### Microscopy ###

```{r load growth data}
growth <- read_tsv("micro_cultures.tsv", col_types = cols(.default = col_character(), 
                                                  count = col_integer(),
                                                  date = col_date(format = "%d-%m-%Y"))
                   )
```


```{r growth plot meteoric}
# Microscopy counts on KR0015 groundwater in duplicates, obtaining 680,220 and 890,910 cells ml-1.
# P26201_1058 KR0015 lysate unfiltered
# P26201_1061 KR0015 srb    unfiltered
# P26201_1067 KR0015 lysate filtered
# P21015_1077 KR0015 srb    filtered

p1 <- growth %>% # KR0015
  filter(sample %in% c("P26201_1058","P26201_1061",
                       "P26201_1067","P21015_1077")) %>%
  ggplot(aes(x = date, y = count)) +
  geom_hline(yintercept = 785565, linewidth = 6, alpha = 0.3) + # t0
  geom_point(aes(colour = fraction)) +
  geom_line(aes(group = sample, colour = fraction), linewidth = 0.5) +
  scale_x_date(limits = c(as.Date("15-10-2021", format = "%d-%m-%Y"),
                          as.Date("15-02-2022", format = "%d-%m-%Y"))) +
  scale_y_log10(limits = c(1e4,1e8),
                breaks = c(1e4,1e5,1e6,1e7),
                labels = c(expression("10"^"4"),
                           expression("10"^"5"),
                           expression("10"^"6"),
                           expression("10"^"7"))
                ) +
  scale_colour_manual(values = c("unfiltered" = "#a6cee3", "filtered" = "#1f78b4"),
                      labels = c("> 0.1 µm","0.1 - 0.45 µm")) +
  labs(x = "", y = expression("Cells [ml"^"-1"*"]"), colour = "Fraction") +
  annotate(geom = "text", 
           x = as.Date("2022-02-06", format = "%Y-%m-%d"), 
           y = 785565, label = "t0", size = 6/.pt, hjust = 0) +
  annotate(geom = "text", 
           x = as.Date("2022-02-06", format = "%Y-%m-%d"), 
           y = 5810000, label = "SRB", size = 6/.pt, hjust = 0) +
  annotate(geom = "text", 
           x = as.Date("2022-02-06", format = "%Y-%m-%d"), 
           y = 19100000, label = "Lys", size = 6/.pt, hjust = 0) +
    annotate(geom = "text", 
           x = as.Date("2022-02-06", format = "%Y-%m-%d"), 
           y = 183000, label = "Lys", size = 6/.pt, hjust = 0) +
    annotate(geom = "text", 
           x = as.Date("2022-02-06", format = "%Y-%m-%d"), 
           y = 67700, label = "SRB", size = 6/.pt, hjust = 0) +
  annotate(geom = "text",x = as.Date("2021-10-15"), y = Inf, 
           label = "a: meteoric", size = 7/.pt, hjust = 0.1, vjust = 2) +
  theme_tidy() + 
  theme(
    plot.margin = margin(0,0,0,0), 
    axis.ticks.y = element_blank(),
    panel.grid.major.y = element_line(linewidth = 0.4, linetype = "dotted", colour = "grey")
    )
```


```{r growth plot marine}
# Microscopy counts on SA1420 groundwater in duplicates, obtaining 59220 and 73910 cells ml-1.
# For SA2600, this number was 14,100 and 23,000 cells ml-1
# For KA3385, this number was similar; 12,800 and 16,300 cells ml-1 
# Microscopy counts on KR0015 groundwater in duplicates, obtaining 680,220 and 890,910 cells ml-1.
# P26201_1026 SA1420 srb    unfiltered
# P26201_1028 SA1420 lysate unfiltered
# P26201_1029 SA1420 lysate filtered
# P26201_1027 SA1420 srb    filtered

p2 <- growth %>% # SA1420
  filter(sample %in% c("P26201_1028","P26201_1026","P26201_1029","P26201_1027")) %>%
  ggplot(aes(x = date, y = count)) +
  geom_hline(yintercept = 66565, size = 5, alpha = 0.3) + # t0
  geom_point(aes(colour = fraction)) +
  geom_line(aes(group = sample, colour = fraction), linewidth = 0.5) +
  scale_x_date(limits = c(as.Date("15-10-2021", format = "%d-%m-%Y"),
                          as.Date("15-02-2022", format = "%d-%m-%Y"))) +
  scale_y_log10(limits = c(1e4,1e8), breaks = c(1e4,1e5,1e6,1e7)) +
  labs(x = "", y = expression(Cells~ml^{-1}), colour = "Fraction") +
  scale_colour_manual(values = c("unfiltered" = "#a6cee3", "filtered" = "#1f78b4"),
                      labels = c("> 0.1 µm","0.1 - 0.45 µm"), guide = "none") +
  annotate(geom = "text", 
           x = as.Date("2022-02-06", format = "%Y-%m-%d"), 
           y = 1800000, label = "Lysate", size = 6/.pt, hjust = 0) +
  annotate(geom = "text", 
           x = as.Date("2022-02-06", format = "%Y-%m-%d"), 
           y = 601000, label = "SRB", size = 6/.pt, hjust = 0) +
  annotate(geom = "text", 
           x = as.Date("2022-02-06", format = "%Y-%m-%d"), 
           y = 92800, label = "SRB", size = 6/.pt, hjust = 0) +
  annotate(geom = "text", 
           x = as.Date("2022-02-06", format = "%Y-%m-%d"), 
           y = 64700, label = "Lysate", size = 6/.pt, hjust = 0) +
  annotate(geom = "text",x = as.Date("2021-10-15"), y = Inf, 
           label = "b: marine", size = 7/.pt, vjust = 2, hjust = 0.1) +
  theme_tidy(legend = "none") +
  theme(axis.text.y = element_blank(), plot.margin = margin(0,0,0,0),
          axis.ticks.y = element_blank(),
          axis.title.y = element_blank(),
          panel.grid.major.y = element_line(linewidth = 0.4, linetype = "dotted", colour = "grey")
        )
```


```{r growth plot saline}
# Microscopy counts on SA2600 groundwater in duplicates, obtaining 14,100 and 23,000 cells ml-1.
# P26201_1008 SA2600 srb unfiltered
# P26201_1051 SA2600 t0

p3 <- growth %>% # SA2600
  filter(sample %in% c("P26201_1008")) %>%
  ggplot(aes(x = date, y = count)) +
  geom_hline(yintercept = 18550, size = 6, alpha = 0.3) + # t0
  geom_point(aes(colour = fraction)) +
  geom_line(aes(group = sample, colour = fraction), size = 0.5) +
  scale_x_date(limits = c(as.Date("15-10-2021", format = "%d-%m-%Y"),
                          as.Date("15-02-2022", format = "%d-%m-%Y"))) +
  scale_y_log10(limits = c(1e4,1e8), breaks = c(1e4,1e5,1e6,1e7)) +
  scale_colour_manual(values = c("unfiltered" = "#a6cee3", "filtered" = "#1f78b4"),
                      labels = c("> 0.1 µm","0.1 - 0.45 µm"),
                      guide = "none") +
  labs(x = "", y = expression(Cells~ml^{-1}), colour = "Fraction") +
  annotate(geom = "text", 
           x = as.Date("2022-02-06", format = "%Y-%m-%d"), 
           y = 221000, label = "SRB", size = 6/.pt, hjust = 0) +
  annotate(geom = "text", 
           x = as.Date("2022-02-06", format = "%Y-%m-%d"), 
           y = 18550, label = "t0", size = 6/.pt, hjust = 0) +
  annotate(geom = "text",x = as.Date("2021-10-15"), y = Inf, 
           label = "c: saline", size = 7/.pt, hjust = 0.1, vjust = 2) +
  theme_tidy(legend = "none") +
  theme(axis.text.y = element_blank(), 
        plot.margin = margin(0,0,0,0),
        axis.ticks.y = element_blank(),
        axis.title.y = element_blank(),
        panel.grid.major.y = element_line(linewidth = 0.4, linetype = "dotted", colour = "grey")
        )
```


```{r plot growth patchwork, fig.width = 18/2.53}
library(patchwork)
p1 + p2 + p3 + plot_layout(guides = "collect") & theme(legend.position = "bottom", plot.background = element_blank())
```

```{r export growth}
ggsave(filename = "figures/growth.pdf", width = 18, height = 8, units = c("cm"))
```

### QPCR KR0015 ###


```{r data qpcr}
read_tsv('data/qpcr.tsv', col_types = cols(.default = col_character(), 
                                      quantity = col_double(),
                                      age = col_factor(levels = seq(1:10) %>% as.character())
                                      )
         ) -> qpcr
```


```{r plot qpcr, fig.width = 6}
qpcr %>%
  group_by(sample, Kingdom) %>% 
  summarise(mean = mean(quantity), sd = sd(quantity), .groups = 'drop') %>%
  inner_join(smd, by = "sample") %>%
  mutate(category = paste(medium, Kingdom)) %>%
  mutate(fraction = if_else(fraction == 'filtered', 
                       '0.1 - 0.45 µm fraction', 
                       '> 0.1 µm fraction')
         ) %>%
  # Call plot
  ggplot(aes(x = fct_relevel(age, seq(1:10) %>% as.character), 
             y = mean, 
             group = category, 
             colour = Kingdom)) +
  geom_line(aes(linetype = medium)) +
  facet_wrap(~ fraction) +
  geom_pointrange(aes(ymin = mean - sd, 
                      ymax = mean + sd), 
                  size = 0.2) +
  labs(x = 'Week', y = expression("Gene copy number [ml"^"-1"*"]")) +
  scale_y_log10(breaks = c(1e3,1e4,1e5,1e6,1e7), 
                labels = c(expression("10"^"3"),
                           expression("10"^"4"),
                           expression("10"^"5"),
                           expression("10"^"6"),
                           expression("10"^"7")
                           )
                ) +
  scale_linetype(name = "Medium:", labels = c("SRB","Cell lysate")) +
  scale_color_manual(name = "", values = c("Archaea" = "#a6cee3", "Bacteria" = "#1f78b4")) +
  theme_tidy(legend = "right") +
  theme(panel.grid.major.y = element_line(colour = "grey", linetype = "dotted", size = 0.5),
        axis.ticks.y = element_blank(),
        strip.background = element_blank())
```


```{r export qpcr}
ggsave(filename = "figures/qpcr.pdf", width = 16, height = 8, units = c("cm"))
```


### Areaplot KR0015 ###

```{r select most abundant asvs, message=FALSE, warning=FALSE}
seqtab %>%
  filter(sample %in% paste("P21015", seq(1042,1081), sep = "_")
           ) %>%
  inner_join(tax, by = "seqid") %>%
  group_by(phylum, sample) %>%
  # Sum the abundance of each phylum within a sample
  summarise(relab = sum(relab), .groups = 'drop_last') %>%
  # Calculate the mean abundance of each phylum over the categories
  summarise(mean_relab = sum(relab), .groups = 'drop') %>%
  filter(!is.na(phylum)) %>%
  top_n(9, mean_relab) -> t
```


```{r add selection to taxonomy table, message=FALSE, warning=FALSE}
tax %>%
  left_join(t %>% transmute(phylum, topphylum = phylum), by = "phylum") %>%
  replace_na(list("topphylum" = "Other")) -> taxref
```


```{r areaplot P21015 phylum}
seqtab %>%
  filter(sample %in% paste("P21015", seq(1042,1081), sep = "_")
           ) %>%
  inner_join(taxref, by = "seqid") %>%
  # Summarize in order to have the sum for each category and topphylum
  group_by(sample, topphylum) %>% 
  summarise(relab = sum(relab), .groups = 'drop') %>%
  inner_join(smd, by = "sample") %>%
  mutate(fraction = if_else(fraction == 'filtered', 
                       '0.1 - 0.45 µm fraction', 
                       '> 0.1 µm fraction')
         ) %>%
  mutate(medium = if_else(medium == 'lys', 
                       'Cell lysate', 
                       'SRB')
         ) %>%
  mutate(fraction = paste(medium, fraction)) %>%
  mutate(age = as.numeric(.$age)) %>% 
  # Call the plot
  ggplot(aes(x = age, 
             y = relab, 
             fill = fct_relevel(topphylum, c("Other")))
         ) +
  geom_area() + scale_x_continuous(n.breaks = 10) +
  facet_wrap(~ fraction) +
  labs(x = 'Week', y = 'Relative abundance') +
  scale_fill_brewer(palette = 'Paired') +
  theme_tidy(fontsize = 7) +
  theme(
    legend.position = "right",
    legend.key.size = unit(4, "mm"),
    strip.text = element_text(size = 7, face = "bold"))
```


```{r export area plot}
ggsave("figures/areaplot_phylum.pdf", height = 14, width = 16, units = "cm")
```


```{r most abundant phyla (t0)}
# Use the samples from P15009 and P21015 as t0
t0 <- c(
  "P15009_1057","P15009_1059","P15009_1060","P21015_1020","P21015_1021","P21015_1022", # KR0015
  "P15009_1061","P15009_1065","P15009_1066","P21015_1023","P21015_1024","P21015_1025", # SA1420
  "P15009_1069","P15009_1062","P15009_1063","P21015_1029","P21015_1030","P21015_1031"  # SA2600
       )

seqtab %>%
  filter(sample %in% t0) %>%
  inner_join(tax, by = "seqid") %>%
  group_by(phylum, sample) %>%
  # Sum the abundance of each phylum within a sample
  summarise(relab = sum(relab), .groups = 'drop_last') %>%
  # Calculate the mean abundance of each phylum over the categories
  summarise(mean_relab = sum(relab), .groups = 'drop') %>%
  filter(!is.na(phylum)) %>%
  top_n(8, mean_relab) -> t

tax %>%
  left_join(t %>% transmute(phylum, topphylum = phylum), by = "phylum") %>%
  mutate(topphylum = if_else(is.na(phylum), "Unidentified", topphylum)) %>%
  replace_na(list("topphylum" = "Other")) -> taxref
```


```{r plot environmental communities (t0), fig.width = 6}
seqtab %>%
  filter(sample %in% t0) %>%
  inner_join(taxref, by = "seqid") %>%
  # Summarize in order to have the sum for each category and topphylum
  group_by(sample, topphylum) %>% 
  summarise(mrelab = sum(relab), .groups = 'drop') %>%
  # Call the plot
  ggplot(aes(x = fct_relevel(sample, t0), 
             y = mrelab * 100, 
             fill = fct_relevel(topphylum, c("Unidentified","Other")))) +
  labs(x = '', y = 'Relative abundance (%)', fill = "Phylum") +
  geom_col() + scale_y_continuous(expand = c(0.01,0)) +
  annotate(geom = "segment", x = 0.55, xend = 6.45, y = -2, yend = -2) +
  annotate(geom = "segment", x = 6.55, xend = 12.45, y = -2, yend = -2) +
  annotate(geom = "segment", x = 12.55, xend = 18.45, y = -2, yend = -2) +
  annotate(geom = "text", x = 3.5, y = -5, label = "KR0015", size = 7/.pt) +
  annotate(geom = "text", x = 9.5, y = -5, label = "SA1420", size = 7/.pt) +
  annotate(geom = "text", x =15.5, y = -5, label = "SA2600", size = 7/.pt) +
  scale_fill_brewer(palette = 'Paired') +
  theme_tidy(ratio = 0.7) + 
  theme(axis.text.x = element_blank(), 
        axis.ticks.x = element_blank(),
        legend.key.size = unit(4, "mm"))
```


```{r export t0 plot}
ggsave("figures/t0_phylum.pdf", height = 10, width = 16, units = "cm")
```


```{r most abundant taxa}
seqtab %>% 
  inner_join(smd) %>% 
  filter(groundwater %in% c("SA1420","KR0015","SA2600") & medium == "env") %>% 
  inner_join(tax) %>% group_by(taxon) %>% 
  summarise(n = sum(relab)) %>% 
  slice_max(n, n = 25) %>%
  pull(taxon) -> t

seqtab %>%
  inner_join(tax, by = "seqid") %>% filter(taxon %in% t) %>%
  group_by(sample, taxon) %>% summarise(relab = sum(relab)*100, .groups = "drop") %>%
  inner_join(smd, by = "sample") %>% 
  filter(groundwater %in% c("SA1420","KR0015","SA2600")) %>%
  filter(relab >= 1) %>%
  ggplot(aes(x = relab, y = fct_relevel(taxon, rev(t)))
         ) +
  geom_boxplot(fill = "#d9d9d9", size = 0.3) +
  scale_x_log10(breaks = c(0.1,1,10,100), labels = c("0.1", "1", "10", "100")) +
  labs(x = "Relative abundance (%)") +
  facet_wrap(~groundwater) +
  theme_tidy(fontsize = 7, ratio = 2) +
  theme(
    axis.title.y = element_blank(),
    axis.ticks.y = element_blank(),
    axis.text.y = element_text(face = "plain"),
    strip.text = element_text(size = 7, face = "bold")
  )
```
```{r export boxplot abundant taxa}
ggsave("figures/boxplot_top.pdf", height = 10, width = 16, units = "cm")
```


```{r rda-metadata}
# t0 are the environmental samples with n = 6 for each groundwater type
t0 <- c(
  "P15009_1057","P15009_1059","P15009_1060","P21015_1020","P21015_1021","P21015_1022", # KR0015
  "P15009_1061","P15009_1065","P15009_1066","P21015_1023","P21015_1024","P21015_1025", # SA1420
  "P15009_1069","P15009_1062","P15009_1063","P21015_1029","P21015_1030","P21015_1031"  # SA2600
       )

geochemistry <- list( # Concentrations are in mg l-1
  groundwater = c("KR0015","SA1420","SA2600"),
  strontium = c(0.716,0.716,0.719),
  sulfate = c(60.3,314,630),
  ph = c(7.4,7.5,7.5),
  ec = c(264,958,3550),
  doc = c(15.4,6.6,1.0),
  oxygen = c(-10.5,-7.45,-12.2),
  nitrogen = c(0.25,1.82,0.08),
  sulfur = c(0.482,0.094,0),
  s34 = c(27.4,30.0,13.1)
  ) %>% data.frame() %>%
  inner_join(smd[,c("sample","groundwater")], by = "groundwater")

geochemistry <- seqtab %>% # Add the species richness for each sample
  group_by(sample) %>% add_tally() %>% ungroup() %>%
  select(sample, n) %>% distinct() %>% inner_join(geochemistry, by = "sample")
```


```{r rda-data}
hellinger <- seqtab %>%
  filter(sample %in% t0) %>%
  # Standard Vegan transformation: Turn table with with samples as *rows*
  select(sample, seqid, count) %>%
  spread(seqid, count, fill = 0) %>%
  # Turn into a numeric matrix
  column_to_rownames('sample') %>% vegan::decostand(method = "hellinger") %>%
  rownames_to_column("sample") %>%
  pivot_longer(-1, names_to = "seqid", values_to = "hellinger")

m <- hellinger %>% # Create a matrix
  spread(seqid, hellinger) %>%
  column_to_rownames("sample")
```


```{r rda}
vegan::rda(
  m ~ oxygen + doc + s34, correlation = F,
  data = geochemistry %>%
    semi_join(hellinger, by = 'sample') %>%
    filter(sample %in% t0) %>%
    arrange(sample) %>%
    column_to_rownames('sample') 
) -> rda

# This one will contain the proportions explained which we get by dividing the
# eigenvalue of each component with the sum of eigenvalues for all components.
p <- c(rda$CCA$eig, rda$CA$eig)/sum(c(rda$CCA$eig, rda$CA$eig))
# This one is collecting the coordinates for arrows that will depict how the 
# factors in our model point in the coordinate
a <- rda$CCA$biplot %>% data.frame() %>% tibble::rownames_to_column('factor')

# sulfate (RDA1, RDA2) = (0.95,0.31)
# ec (1.0, 0.05)
# nitrogen (-0.45, 0.89)
# s34 (-0.95,0.28)
a <- list(factor = c("s34"),
          RDA1 = c(-0.96),
          RDA2 = c( 0.29)) %>% data.frame() %>% rbind(a)
```


```{r rda-plot, fig.width = 10/2.53}
rda$CCA$wa %>% data.frame() %>% tibble::rownames_to_column('sample') %>%
  inner_join(geochemistry, by = 'sample') %>%
  ggplot(aes(x = RDA1, y = RDA2)) +
  geom_vline(xintercept = 0, colour = "grey", linetype = "dotted") +
  geom_hline(yintercept = 0, colour = "grey", linetype = "dotted") +
  geom_point(aes(colour = groundwater, size = n), shape = 1, stroke = 1) +
  xlab(sprintf("RDA1 (%2.1f%% explained)", p[[1]] * 100)) +
  ylab(sprintf("RDA2 (%2.1f%% explained)", p[[2]] * 100)) +
  geom_segment(
    data = a, mapping = aes(xend = RDA1/4, yend = RDA2/4), 
    x = 0, y = 0, arrow = arrow(length = unit(2, "mm"), type = "closed") 
    ) +
  geom_text(
    data = a, mapping = aes(x = RDA1/8, y = RDA2/8, label = factor), 
    size = 7/.pt
    ) + scale_size(range = c(1,8), name = "") +
  annotate("text", x = -0.1, y =-0.25, label = "KR0015", colour = cols.gw[1], size = 7/.pt) +
  annotate("text", x = 0.05, y = 0.28, label = "SA1420", colour = cols.gw[2], size = 7/.pt) +
  annotate("text", x =  0.2, y =-0.05, label = "SA2600", colour = cols.gw[3], size = 7/.pt) +
  stat_ellipse(aes(x = RDA1, y = RDA2, colour = groundwater), 
               geom = "polygon", show.legend = F,
               fill = NA, linetype = "dashed") +
  scale_color_manual(name = "",values = cols.gw, guide = "none") +
  scale_fill_manual(name = "",values = cols.gw, guide = "none") +
  theme_tidy(legend = "right")
```
```{r export rda}
ggsave("figures/rda.pdf", height = 10, width = 12, units = "cm")
```


```{r rarefaction}
seqtab %>%
  inner_join(smd, by = "sample") %>% filter(groundwater == "KR0015B") %>%
  select(sample, seqid, count) %>%
  spread(seqid, count, fill = 0) %>%
  column_to_rownames('sample') -> seqtab_matrix
  
png(filename = "figures/rarecurve_meteoric.png", width = 14, height = 10, units = "cm", res = 300)  
min(rowSums(seqtab_matrix)) %>%
vegan::rarecurve(seqtab_matrix, 
                 sample = ., step = 100, cex = 0.5, cex.axis = 0.6, cex.lab = 0.6, 
                 xlab = 'Sequencing depth KR0015B', 
                 ylab = 'Rarefied No. of ASVs', label = TRUE)
dev.off()


seqtab %>%
  inner_join(smd, by = "sample") %>% filter(groundwater == "SA1420A-1") %>%
  select(sample, seqid, count) %>%
  spread(seqid, count, fill = 0) %>%
  column_to_rownames('sample') -> seqtab_matrix
  
png(filename = "figures/rarecurve_mm.png", width = 14, height = 10, units = "cm", res = 300)  
min(rowSums(seqtab_matrix)) %>%
vegan::rarecurve(seqtab_matrix, 
                 sample = ., step = 100, cex = 0.5, cex.axis = 0.6, cex.lab = 0.6, 
                 xlab = 'Sequencing depth SA1420A-1', 
                 ylab = 'Rarefied No. of ASVs', label = TRUE)
dev.off()


seqtab %>%
  inner_join(smd, by = "sample") %>% 
  filter(groundwater %in% c("KA3511A-1","SA2600A-1","KA2862A-1")) %>%
  select(sample, seqid, count) %>%
  spread(seqid, count, fill = 0) %>%
  column_to_rownames('sample') -> seqtab_matrix
  
png(filename = "figures/rarecurve_os.png", width = 14, height = 10, units = "cm", res = 300)  
min(rowSums(seqtab_matrix)) %>%
vegan::rarecurve(seqtab_matrix, 
                 sample = ., step = 100, cex = 0.5, cex.axis = 0.6, cex.lab = 0.6, 
                 xlab = 'Sequencing depth SA2600, KA3511 & KA2862', 
                 ylab = 'Rarefied No. of ASVs', label = TRUE)
dev.off()
```

